modules/gtk/hm.nix

```
@@ -29,25 +28,70 @@ in {
         window.background { border-radius: 0; }
       '';
     };
+
+    flatpakSupport.enable =
+      config.lib.stylix.mkEnableTarget "support for theming Flatpak apps" true;
   };
 
-  config = lib.mkIf cfg.enable {
-    # programs.dconf.enable = true; required in system config
-    gtk = {
-      enable = true;
-      font = {
-        inherit (config.stylix.fonts.sansSerif) package name;
-        size = config.stylix.fonts.sizes.applications;
+  config = lib.mkIf cfg.enable (lib.mkMerge [
+    {
+      # programs.dconf.enable = true; required in system config
+      gtk = {
+        enable = true;
+        font = {
+          inherit (config.stylix.fonts.sansSerif) package name;
+          size = config.stylix.fonts.sizes.applications;
+        };
+        theme = {
+          package = pkgs.adw-gtk3;
+          name = "adw-gtk3";
+        };
       };
-      theme = {
-        package = pkgs.adw-gtk3;
-        name = "adw-gtk3";
+
+      xdg.configFile = {
+        "gtk-3.0/gtk.css".source = finalCss;
+        "gtk-4.0/gtk.css".source = finalCss;
       };
-    };
+    }
 
-    xdg.configFile = {
-      "gtk-3.0/gtk.css".source = finalCss;
-      "gtk-4.0/gtk.css".source = finalCss;
-    };
-  };
+    (lib.mkIf cfg.flatpakSupport.enable (lib.mkMerge [
+      {
+        # Flatpak apps apparently don't consume the CSS config. This workaround appends it to the theme directly.
+        home.file.".themes/${config.gtk.theme.name}".source = pkgs.stdenvNoCC.mkDerivation {
+          name = "flattenedGtkTheme";
+          src = "${config.gtk.theme.package}/share/themes/${config.gtk.theme.name}";
+
+          installPhase = ''
+            cp --recursive . $out
+            cat ${finalCss} | tee --append $out/gtk-{3,4}.0/gtk.css
+          '';
+        };
+      }
+      (
+        let
+          filesystem = "${config.home.homeDirectory}/.themes/${config.gtk.theme.name}:ro";
+          theme = config.gtk.theme.name;
+        in
+          if lib.hasAttrByPath ["services" "flatpak" "overrides"] options
```

> `hasAttr` returns `true` if _set_ has an attribute named _s_, and `false` otherwise. This is a dynamic version of the `?` operator, since _s_ is an expression rather than an identifier.
>
> Nix, `builtins.hasAttr`

Can we not simplify this to:

```suggestion
          if options ? services.flatpak.overrides
```

---

Yes, you are right! I didn't know the builtin `?` operator could handle nested attribures. The documentation for the `lib.hasAttrByPath` points this out as well: https://nixos.org/manual/nixpkgs/unstable/#function-library-lib.attrsets.hasAttrByPath